guidelines/REUSE.md

```diff
@@ -1,38 +1,65 @@
 # Reuse Philosophy
+
 We encourage reuse and patterns through references.
 
 ## What is reusable
+
 The following types are reusable, as defined by the spec:
 
-* Operations
 * Parameters
-* Responses
 * Models (or Schema Objects in general)
+* Responses
+* Operations (_Operations can only be referenced externally_)
 
 ## Reuse strategy
+
 When reusing components in an API design, a pointer is created from the definition to target design.  The references are maintained in the structure, and can be updated by modifying the source definitions.  This is different from a "copy on design" approach where references are injected into the design itself.
 
 The reuse technique is transparent between JSON or YAML and is lossless when converting between the two.
 
-YAML anchors are technically allowed but break the general reuse strategy in Swagger, since anchors are "injected" into a single document.  They are not recommended.
+YAML anchors are technically allowed but break the general reuse strategy in OpenAPI Specification, since anchors are "injected" into a single document.  They are not recommended.
 
-Referenes can be made either inside the Swagger definition file or to external files. References are done using [JSON Reference](http://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03).
+Referenes can be made either internal to the OpenApi Specification file or to external files. 
 
 ## Techniques
 
 ### Guidelines for Referencing
 
+Whether you reference definitions internally or externally, you can never override or change their definitions from the referring location. The definitions can only be used as-is.
+
+#### Internal references
+
 When referencing internally, the target references have designated locations:
 
-* Parameters -> `parameters`
-* Responses -> `responses`
-* Models (and general Schema Objects) -> `definitions`
+* Parameters -> `#/parameters`
+* Responses -> `#/responses`
+* Models (and general Schema Objects) -> `#/definitions`
+
+All references are canonical and must be a qualified [JSON Pointer, per RFC6901](http://tools.ietf.org/html/rfc6901). For example, simply referencing a model `Pet` is not allowed, even if there are no other definitions of it in the file.
+
+_Example from https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v2.0/json/petstore.json_
```

Comment:
Didn't really go into the details of the containers. There are two reasons for the containers (that I recall). 1 - allow for easier validation. Since you *know* that anything under `definitions` is a Schema Object you can easily validate them. To me, that's a major downside of external references as it makes validating a spec more difficult (not getting into the why now). 2 - All spec objects have no `additionalProperties` allowed except for vendor extensions. That again is intentional, and that's another reason to provide the internal containers to hold referenced objects, because those can't just be added to the spec arbitrarily.